scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/KryoCoder.scala

```diff
@@ -0,0 +1,28 @@
+package com.twitter.scalding.beam_backend
+
+import com.esotericsoftware.kryo.io.Input
+import com.twitter.chill.{ KryoInstantiator, KryoPool }
+import java.io.{ InputStream, OutputStream }
+import org.apache.beam.sdk.coders.AtomicCoder
+import scala.language.implicitConversions
+
+final class KryoCoder(kryoInstantiator: KryoInstantiator) extends AtomicCoder[Any] {
+  @transient lazy val kryoPool = KryoPool.withByteArrayOutputStream(Runtime
+    .getRuntime
+    .availableProcessors, kryoInstantiator)
+  override def encode(value: Any, os: OutputStream): Unit = {
+    val bytes = kryoPool.toBytesWithClass(value)
+    os.write(bytes.length)
```

Comment:
this isn't safe... this can only write 1 byte (even though it takes an int). so if you have more than 255 bytes of data it will fail.

See:
https://github.com/twitter/chill/blob/19daefccbcda7d50f3130985de6e016be6eafa01/chill-hadoop/src/main/java/com/twitter/chill/hadoop/KryoDeserializer.java#L49

you might want to use this class:
https://github.com/twitter/scalding/blob/b6d465f6fc308d66e5a53c62662d5860846d8dc4/scalding-serialization/src/main/scala/com/twitter/scalding/serialization/JavaStreamEnrichments.scala#L238

which can write positive ints to streams for you (and read them).

We use this class already for OrderedSerialization work.